applications/fishing-map/src/features/download/downloadTrack.slice.ts

```diff
@@ -0,0 +1,126 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { DateTime } from 'luxon'
+import { stringify } from 'qs'
+import { saveAs } from 'file-saver'
+import { DownloadActivity } from '@globalfishingwatch/api-types'
+import GFWAPI from '@globalfishingwatch/api-client'
+import { RootState } from 'store'
+import { AsyncError, AsyncReducerStatus } from 'utils/async-slice'
+import { DateRange } from 'features/analysis/analysis.slice'
+import { Format } from './downloadTrack.config'
+
+type VesselParams = {
+  name: string
+  id: string
+  datasets: string
+}
+
+export interface DownloadTrackState {
+  name: string
+  id: string
+  datasets: string
+  status: AsyncReducerStatus
+}
+
+const initialState: DownloadTrackState = {
+  name: '',
+  id: '',
+  datasets: '',
+  status: AsyncReducerStatus.Idle,
+}
+
+export type DownloadTrackParams = {
+  vesselId: string
+  vesselName: string
+  dateRange: DateRange
+  datasets: string
+  format: Format | 'lines'
+}
+
+export const downloadTrackThunk = createAsyncThunk<
+  DownloadActivity,
+  DownloadTrackParams,
+  {
+    rejectValue: AsyncError
+  }
+>('download/create', async (params: DownloadTrackParams, { rejectWithValue }) => {
```

Comment:
⚠️ this is using the same action name than the `downloadActivityThunk` which will have unexpected behaviour for sure, please update them

Reply:
good catch